### PR TITLE
feat: Expose hasString method for iOS 14+

### PR DESCRIPTION
**Pull request by gnardini**
Firstly, thanks for the work on the clipboard!

# Overview
iOS 14 now shows a notification each time an app accesses the clipboard, so we need to be more careful on when to read it. It also has a new API that allows apps to tell if there's any content on the clipboard or not. The API has more methods than the one I implemented, I just added the one I needed. I don't have time to keep working on this unfortunately to add the rest of the methods :( I understand if you think this is not enough to merge.

# Test Plan
Unfortunately, the example project didn't compile for me, so I couldn't use that one to test, I tested directly on my own project.
To test you need an iOS 14 device and one with an older version. On both of them add a call somewhere to `await Clipboard.hasString()`. Do so while having content on the clipboard and while having an empty clipboard. You could print the result of the call to `console.log` or any other logger to see that the correct value is returned.

### CODE REVIEW

File: ios/RNCClipboard.m [hunk 1]
```diff
@@ -28,4 +28,19 @@ - (dispatch_queue_t)methodQueue
   resolve((clipboard.string ? : @""));
 }
 
+RCT_EXPORT_METHOD(hasString:(RCTPromiseResolveBlock)resolve
+                  reject:(__unused RCTPromiseRejectBlock)reject)
+{
+  bool stringPresent = false;
```
Comment (jeanregisser) [inline ios/RNCClipboard.m:34]: nit: `BOOL` should be preferred in ObjC, with `YES` or `NO` as values.

See https://stackoverflow.com/questions/541289/objective-c-bool-vs-bool

File: ios/RNCClipboard.m [hunk 2]
```diff
@@ -28,4 +28,19 @@ - (dispatch_queue_t)methodQueue
...
+  UIPasteboard *clipboard = [UIPasteboard generalPasteboard];
+  if (@available(iOS 10, *)) {
+    stringPresent = clipboard.hasStrings;
+  } else {
+    NSString* stringInPasteboard = clipboard.string;
+    stringPresent = stringInPasteboard != nil;
```
Comment (jeanregisser) [inline ios/RNCClipboard.m:40]: Can it be empty too?

**Comment by filiptronicek** [comment]
Awesome! Can we implement stuff from https://github.com/react-native-community/clipboard/issues/64 also?

**Review by Naturalclar — APPROVED** [review]
@gnardini amazing! tested on example app and seems to be working great.
sorry for the example project on this repo not working for you, there was a typo in podfile which was preventing the example to work, but now it's fixed.

**Comment by barrymcgee** [comment]
I'm finding that `await Clipboard.hasString()` also returns as true even after running `pbcopy < /dev/null` - is this expected?

https://github.com/react-native-clipboard/clipboard/issues/125

**Pull request merged**